pkg/controller/che/che_controller.go

```diff
@@ -466,69 +468,52 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 	externalDB := instance.Spec.Database.ExternalDb
 	if !externalDB {
 		if cheMultiUser == "false" {
-			if k8sclient.IsDeploymentExists("postgres", instance.Namespace) {
-				k8sclient.DeleteDeployment("postgres", instance.Namespace)
+			if k8sclient.IsDeploymentExists(deploy.PostgresDeploymentName, instance.Namespace) {
+				k8sclient.DeleteDeployment(deploy.PostgresDeploymentName, instance.Namespace)
 			}
 		} else {
+			postgresLabels := deploy.GetLabels(instance, deploy.PostgresDeploymentName)
+
 			// Create a new postgres service
-			postgresLabels := deploy.GetLabels(instance, "postgres")
-			postgresService := deploy.NewService(instance, "postgres", []string{"postgres"}, []int32{5432}, postgresLabels)
-			if err := r.CreateService(instance, postgresService, false); err != nil {
-				return reconcile.Result{}, err
-			}
-			// Create a new Postgres PVC object
-			pvc := deploy.NewPvc(instance, deploy.DefaultPostgresVolumeClaimName, "1Gi", postgresLabels)
-			if err := r.CreatePVC(instance, pvc); err != nil {
-				return reconcile.Result{}, err
-			}
+			serviceStatus := deploy.SyncServiceToCluster(instance, "postgres", []string{"postgres"}, []int32{5432}, postgresLabels, clusterAPI)
 			if !tests {
-				err = r.client.Get(context.TODO(), types.NamespacedName{Name: pvc.Name, Namespace: instance.Namespace}, pvc)
-				if pvc.Status.Phase != "Bound" {
-					k8sclient.GetPVCStatus(pvc, instance.Namespace)
+				if !serviceStatus.Continue {
+					logrus.Info("Waiting on service 'postgres' to be ready")
+					if serviceStatus.Err != nil {
+						logrus.Error(serviceStatus.Err)
+					}
+
+					return reconcile.Result{Requeue: serviceStatus.Requeue}, serviceStatus.Err
 				}
 			}
-			// Create a new Postgres deployment
-			postgresDeployment := deploy.NewPostgresDeployment(instance, isOpenShift, cheFlavor)
 
-			if err := r.CreateNewDeployment(instance, postgresDeployment); err != nil {
-				return reconcile.Result{}, err
-			}
-			time.Sleep(time.Duration(1) * time.Second)
-			pgDeployment, err := r.GetEffectiveDeployment(instance, postgresDeployment.Name)
-			if err != nil {
-				logrus.Errorf("Failed to get %s deployment: %s", postgresDeployment.Name, err)
-				return reconcile.Result{}, err
-			}
+			// Create a new Postgres PVC object
+			pvcStatus := deploy.SyncPVCToCluster(instance, deploy.DefaultPostgresVolumeClaimName, "1Gi", postgresLabels, clusterAPI)
 			if !tests {
-				if pgDeployment.Status.AvailableReplicas != 1 {
-					scaled := k8sclient.GetDeploymentStatus("postgres", instance.Namespace)
-					if !scaled {
-						return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
+				if !pvcStatus.Continue {
+					logrus.Infof("Waiting on pvc '%s' to be bound", deploy.DefaultPostgresVolumeClaimName)
+					if pvcStatus.Err != nil {
+						logrus.Error(pvcStatus.Err)
 					}
+
+					return reconcile.Result{Requeue: pvcStatus.Requeue, RequeueAfter: time.Second * 10}, pvcStatus.Err
```

Comment:
Same remark, maybe 10 seconds is a bit too much ? (or maybe not ;-) )

Reply:
Changed to 1 sec